managed/services/telemetry/deps.go

```diff
@@ -47,6 +47,8 @@ type DataSourceLocator interface {
 
 // DataSource telemetry data source.
 type DataSource interface {
+	PreFetch(ctx context.Context) error
 	FetchMetrics(ctx context.Context, config Config) ([][]*pmmv1.ServerMetric_Metric, error)
+	PostFetch(ctx context.Context) error
```

Comment:
nit: I'd call it `AfterFetch`, because `PostFetch` is a bit less clear

Reply:
Changed.